src/freeze_model.py

```diff
@@ -42,122 +46,128 @@ def strip(input_graph, drop_scope, input_before, output_after, pl_name):
     # which experiment we want to evaluate?
     # Use the -l functionality to ensamble models: python arg.py -l 1234 2345 3456 4567
     parser = argparse.ArgumentParser()
-    parser.add_argument('model_fol')
-    parser.add_argument('-l', '--list', nargs='+', help='List of models to evaluate', required=True)
+    parser.add_argument('exp_dir')
 
     args = parser.parse_args()
-    models = args.list
-    model_fol = args.model_fol
-
-    for model in models:
-        experiment_folder = os.path.join(model_fol, str(model))
-        config = json.load(open(os.path.join(experiment_folder, 'config.json')))
-
-        # parser architecture:
-        if config['model_number'] == 2:
-            arch = 'vgg'
-        elif config['model_number'] == 11:
-            arch = 'musicnn'
-        elif config['model_number'] == 20:
-            arch = 'vggish'
-        else:
-            raise Exception('Model number not found ({})'.format(config['model_number']))
-
-        # parser source task:
-        if 'MSD' in config['load_model']:
+    exp_dir = Path(args.exp_dir)
+
+    config = json.load(open((exp_dir / 'config_whole.json').resolve()))
+
+    # get the architecture
+    if config['config_train']['model_number'] == 2:
+        arch = 'vgg'
+    elif config['config_train']['model_number'] == 11:
+        arch = 'musicnn'
+    elif config['config_train']['model_number'] == 20:
+        arch = 'vggish'
+    else:
+        raise Exception('Model number not found ({})'.format(config['config_train']['model_number']))
```

Comment:
I think this should be a more specific exception, e.g. [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError)